### ISSUE TITLE
[Bug] `--cwd` gets ignored when specified inside package.json script

### ISSUE THREAD

**Issue by Knagis** [yarnpkg/berry#2346]
- [ ] I'd be willing to implement a fix

**Describe the bug**

```
{
  "scripts": {
    "foo": "yarn --cwd .yarn exec cd"
  }
}
```

Running `yarn foo` prints out the current path, not `.yarn` as specified in `--cwd`.

Running `yarn --cwd .yarn exec cd` directly in the shell produces the correct output.

**To Reproduce**

Use the following package.json
```
{
  "name": "foo",
  "version": "1.0.0",
  "license": "MIT",
  "scripts": {
    "foo": "yarn --cwd .yarn exec cd"
  }
}
```

Run `yarn foo` from the package directorty.

**Environment if relevant**

 - OS: Windows 10 v2004
 - Node version: 12.13.1
 - Yarn version: 2.4.0

**Comment by arcanis** [comment]
That's expected, `exec` runs from the closest package, ie the root. No idea why you'd want to `--cwd .yarn` 🤔

**Comment by Knagis** [comment]
`--cwd .yarn` was just an example.

The actual usecase in our repo (being migrated from yarn 1) is 

`"install-related": "yarn --cwd ../another-package install"`

we use this because when one package depends on another in a monorepo to install those related dependencies.

Another example:

`"test": "jest && yarn --cwd packages/foo test"`

**Comment by Knagis** [comment]
@arcanis - though if you say it is expected - why is there a difference in behavior when the command is launched directly from the shell and when it is specified in scripts?

**Comment by arcanis** [comment]
I'm confused why you'd do that - aren't you basically reimplementing [workspaces](https://yarnpkg.com/features/workspaces)?

That aside, I was wrong in my previous message (`exec` doesn't actually crawl up the fs hierarchy), but there is still no problem I can see:

```
❯ /tmp.nux6iMkV5S/var/folders/2s/1qvdx9f97plgcgy4r6mr26x80000gn/T/tmp.nux6iMkV5S ❯ yarn --cwd .yarn exec pwd

/private/var/folders/2s/1qvdx9f97plgcgy4r6mr26x80000gn/T/tmp.nux6iMkV5S/.yarn
```

**Comment by Knagis** [comment]
We are reimplementing them (kind of), since we have used yarn1 for a long time. In migration to yarn2 the first step is to make it work as is and only then work on enabling workspaces (and pnp hopefully afterwards).

as for the example: yes, running the command from shell produces the correct output. However, if the exact same command is specified in package.json script, the `--cwd` arg there no longer has any effect.

**Comment by merceyz** [comment]
You can replicate that setup if you set the workspaces field in the root and set [`nmHoistingLimits: workspaces`](https://yarnpkg.com/configuration/yarnrc#nmHoistingLimits) in `.yarnrc.yml`, except for the `yarn.lock` moving to the root it will behave the same

**Comment by Knagis** [comment]
That would only solve install, not running other tasks though.

**Comment by merceyz** [comment]
This is a valid issue, you're running into https://github.com/yarnpkg/berry/pull/1812#discussion_r485873759, as a workaround you can specify the path as the first argument
```
yarn ../another/package exec pwd
```

**Comment by Knagis** [comment]
Thank you! That workaround seems to be working.

**Comment by Mithras** [comment]
I don't think `--cwd` is doing anything at all. It can be confirmed with a simple `yarn --cwd /expected exec pwd`. Output would be anything but `/expected`. What exactly `--cwd` is supposed to be doing?

**Comment by azanatta** [comment]
Hi, having the same issue after the upgrade to berry, every script with cwd stopped working.
When using yarn --cwd ../otherfolder/ valid-script-in-that-package.json-file, yarn runs the script in the current folder and completely ignores the cwd command to change folder for this specific script.

The workaround of removing cwd and simply setting the folder where the script resides, works:
yarn ../otherfolder/ valid-script-etc

**Comment by clemyan** [comment]
I believe this is fixed via #5600
